sql/src/main/java/io/crate/execution/dml/upsert/GeneratedColumns.java

```diff
@@ -126,12 +127,13 @@ public void validateValue(Reference target, Object providedValue, boolean valida
             return;
         }
 
-        Input<?> input = toInject.get(target);
+        // remove the reference from the 'toInject' map
```

Comment:
How about something like:

```
// `toInject` contains columns that might be provided by the user because for nested columns we don't know if the user is going to provide them based on the statement. E.g. in (INSERT INTO (obj) (select ...)) (We know `obj` is going to be provided, but not which nested columns the values will contain.). 
// -> We don't need to inject values that are already provided, so we remove them on the fly.

Reply:
E.g. 

```
INSERT INTO t (obj) (SELECT col1 FROM unnest([{a=1, b=2}, {a=2}]);
```